mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -1767,24 +1775,53 @@ createCPUConvWithMLIR(ModuleOp module, func::FuncOp &func,
         layout = genConfig.inputLayout;
       else
         layout = genConfig.outputLayout;
+      auto direction = genConfig.operation.value();
       for (auto c : layout) {
-        auto direction = genConfig.operation.value();
         if ((direction == rock::ConvOpType::Fwd ||
              direction == rock::ConvOpType::BwdWeight) &&
             tensor == INPUT) {
-          if (c == 'h') {
+          if (c == '0') {               // +++pf: may need adjustment to h/w.
```

Comment:
This whole section needs to be reworked.  It's connecting loopIVs by finding the right letter in the layout string, and needs to have both y/x and h/w present, which causes a collision since 0/1 could mean either.  At the moment, the string will have both 0/1 and y/x, but the better solution is to use a vector of strings like "0i" to allow for unambiguous keys.

There's also logic to override the key match in certain cases, which is necessary to make some test cases work but is confused a bit by the key renaming.  The set of changes here works but is not the final answer.